api/query/query.go

```diff
@@ -61,6 +61,30 @@ func (qh queryHandler) processInput(w http.ResponseWriter, r *http.Request) (*sh
 	return &filters, testRuns, summaries, nil
 }
 
+func (qh queryHandler) getVersionFiles(w http.ResponseWriter, r *http.Request) ([]string, error) {
+
+	filters, err := shared.ParseQueryFilterParams(r.URL.Query())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return nil, err
+	}
+	testRuns, filters, err := qh.getRunsAndFilters(filters)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
```

Comment:
This is a style thing but it's usually safer to write to the `w http.ResponseWriter` in the initial handler function that receives the request. ResponseWriter should only be written to once in the request's life cycle. When a responseWriter is written to in a sub function, it makes it hard to know if responseWriter has already been written to. It's better to raise the error up and let the handler function write it. otherwise, you'll need to let other developers know not to write to responseWriter again.

Reply:
I see. So just to confirm I understand, it's best to pass back the error and handle the error writing in the calling function?

Reply:
Correct!